creator-node/src/dbManager.js

```diff
@@ -166,6 +167,38 @@ class DBManager {
     return error
   }
 
+  /**
+   * Gets a user's files (Files table storagePath column) with pagination.
+   * @param {string} cnodeUserUUID the UUID of the user to fetch file paths for
+   * @param {string} prevStoragePath pagination token (where storagePath > prevStoragePath)
+   * @param {number} batchSize the pagination size (number of file paths to return)
+   * @returns {string[]} user's storagePaths from db
+   */
+  static async getCNodeUserFilesFromDb(
+    cnodeUserUUID,
+    prevStoragePath,
+    batchSize
+  ) {
+    const files = (
+      await models.File.findAll({
+        attributes: ['storagePath'],
+        where: {
+          [sequelize.Op.and]: [
+            sequelize.literal(`"multihash" IN
+          (SELECT "multihash" FROM "Files" WHERE "cnodeUserUUID" = '${cnodeUserUUID}' AND "storagePath" > '${prevStoragePath}')`)
+          ]
+        },
+        group: 'storagePath',
+        having: sequelize.literal(`COUNT(DISTINCT("cnodeUserUUID")) = 1`),
```

Comment:
were you able to confirm that the underlying query here when run uses indexes / isn't too slow? not really worried about the step taking a long time, more just about holding db conn for long time
again can handle separately

Reply:
yeah so I ran manually on prod CN1 to get some timings and since we order and group by `storagePath`, I'm pretty sure it's still going to be faster if we can add an index on `storagePath` and just run the whole query with that instead of also needing `multihash` in the subquery (since `multihash` already has an index). I have a note to look into adding this index as a followup